tools/songAdd.php

```diff
@@ -11,10 +11,8 @@
 		exit("Invalid captcha response");
 
 	$result = $gs->songReupload($_POST['songlink']);
-	if($result == "-4"){
+	if($result == "-4")
 		echo "This URL doesn't point to a valid audio file.";
-	}elseif($result == "-3")
-		echo "This song already exists in our database.";
```

Comment:
If the song exists in the database, it still returns the "song uploaded" message with the ID. It may be confusing. Consider adding "song already exists" but still returning the ID of the already uploaded song.

Reply:
I also thought about this, but the problem is about concept of this "API". It expects to return an integer where if > 0 then it is id, if <= 0 assume it failed. We should ask @Cvolton about designing this

Reply:
It still removes the capability of the error text "This song already exists in our database." Yes, your code still returns the ID of the song, very well but removing the existing error still be confusable.